graftm/hmmer.py

```diff
@@ -598,63 +592,87 @@ def p_search(self, files, args, run_stats, base, unpack, raw_reads):
+            # Extract the orfs of these reads that hit the original search
+            self.extract_orfs(hit_reads_fasta,
+                              extracting_orfm,
+                              hit_reads_orfs_fasta)
+            
+            hit_reads_fasta=hit_reads_orfs_fasta
+        
+        ## TODO: This method does not work when searching a genome, or assembled
+        ## data, because there may be > 1 copy of gene, and very spaced. Simply
+        ## choosing one would not suffice in that context. Will have to rewrite
+        ## So that multiple hits with same read name are delt with properly.
+        #if args.input_sequence_type == 'nucleotide':
+        #    # Only accept 1 HSP per protein sequence
+        #    old_stats = run_stats['reads']
+        #    read_stats = {}
+        #    for orf_name, hsps in old_stats.iteritems():
+        #        read_stats[orf_name] = [hsps[0]]
+        #    run_stats['reads'] = read_stats
         
         # Extract the hits form the original raw read file
-        run_stats['reads'], hit_reads = self.extract_from_raw_reads(files.fa_output_path(base),
-                                                                    hit_readnames,
-                                                                    raw_reads,
-                                                                    unpack.format(),
-                                                                    run_stats['reads'])
-        
-        if args.input_sequence_type == 'nucleotide':
-            # Extract the orfs of these reads that hit the original search
-            hit_orfs = self.extract_orfs(hit_reads,
-                                         files.orf_output_path(base),
-                                         files.orf_hmmsearch_output_path(base),
-                                         files.orf_titles_output_path(base),
-                                         extracting_orfm,
-                                         files.orf_fasta_output_path(base))
-        elif args.input_sequence_type == 'protein':
-            hit_orfs = hit_reads
-        else:
-            raise Exception('Programming Error')
-        # Define the average read length of the hits
-        run_stats['read_length'] = self.check_read_length(hit_orfs, "P")
-        # Stop and log search timer
-        stop = timeit.default_timer()
-        run_stats['search_t'] = str(int(round((stop - start), 0)) )
-        # Falsify some summary entries
-        run_stats['euk_contamination'] = 'N/A'
-        run_stats['euk_uniq'] = 'N/A'
-        run_stats['euk_check_t'] = 'N/A'
-        # Return hit reads, and summary hash
-        return hit_orfs, run_stats
-
-    def d_search(self, files, args, run_stats, base, unpack, raw_reads, euk_check):
-        '''Nucleotide search pipeline - The searching step for the nucleotide
+        hit_read_counts=[0,len(hit_readnames)]
+        return hit_reads_fasta, search_result, hit_read_counts
```

Comment:
It seems a bit clunky to return several things, where one of those is several things. Perhaps return an instance of a new class that includes these info?